Hakawai/Core/TextKit/HKWLayoutManager.m

```diff
@@ -90,8 +90,17 @@ - (NSArray *)roundedRectBackgroundAttributeTuplesInTextStorage:(NSTextStorage *)
     if (!textStorage || range.location == NSNotFound) {
         return nil;
     }
+
+    // Avoid out of bounds crashes that can happen when typing in languages like Tamil
+    NSRange adjustedRange;
+    if (range.location + range.length > textStorage.length) {
+        adjustedRange = NSMakeRange(range.location, textStorage.length - range.location);
```

Comment:
could make this happen every time, rather than only when the ranges have a mismatch. Don't feel strongly about it, just something to think about.